bats/tests/preferences/generate-reg-output.bats

```diff
@@ -0,0 +1,63 @@
+load '../helpers/load'
+
+# No rancher-desktop needed for this test
+local_setup() {
+    if is_windows; then
+        # We need to use a directory that exists on the Win32 filesystem
+        # so the ctrctl clients can correctly map the bind mounts.
+        TEMP="$(win32env TEMP)"
+    else
+        TEMP=/tmp
+    fi
+}
+
+# Bats linter doesn't know that this function calls assert_success
+# so we need to do that explicitly between a `run` line and a call to this function.
+verify_registry_output() {
+    assert_output --partial '[HKEY_CURRENT_USER\SOFTWARE\Policies\Rancher Desktop\defaults\containerEngine]'
+    assert_output --partial '[HKEY_CURRENT_USER\SOFTWARE\Policies\Rancher Desktop\defaults\containerEngine\allowedImages]'
+    assert_output --partial '[HKEY_CURRENT_USER\SOFTWARE\Policies\Rancher Desktop\defaults\containerEngine\allowedImages]'
+    assert_output --partial '"patterns"=hex(7):61,00,62,00,63,00,00,00,67,00,68,00,69,00,00,00,64,00,65,00,66,00,00,00,00,00'
+    assert_output --partial '[HKEY_CURRENT_USER\SOFTWARE\Policies\Rancher Desktop\defaults\WSL\integrations]'
+    assert_output --partial '"first"=dword:1'
+    assert_output --partial '"second"=dword:0'
+}
+
+@test 'generates registry output from stdin' {
+    run bash -c $'echo \'{"kubernetes": {"enabled": false}, "containerEngine": { "allowedImages": {"patterns": ["abc", "ghi", "def"] } }, "WSL": { "integrations": { "first": true, "second": false } } }\' | rdctl create-profile --output reg --hive=hkcu --input - | cat -n'
```

Comment:
I would move the data into a separate function because you are re-using it several times:

```
json_date() {
    echo '{"kubernetes": {"enabled": false}, "containerEngine": { "allowedImages": {"patterns": ["abc", "ghi", "def"] } }, "WSL": { "integrations": { "first": true, "second": false } } }'
}
```

And then you could run it as

```suggestion
    run bash -c 'json_data | rdctl create-profile --output reg --hive=hkcu --input -'
```

I'm assuming the `| cat -n` was leftover from debugging, as the other test cases don't do this.

Reply:
There's no duplication of input after moving to `create-profile-output.bats`, and I am using `cat -n` to verify output ordering of reg files. I don't care about the exact line numbers, just that certain lines appear after others, like children appear after parent lines.

Reply:
Also I don't have access to `json_data` in a bash pipeline, because shell functions don't get exported to child processes.

Reply:
But I don't need the line-numbers if I can test on a here-doc, so I might as well work on that instead.